ServerCore/Pages/Components/PlayerClassSelectComponent.razor.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components;
+using ServerCore.DataModel;
+using ServerCore.Helpers;
+using ServerCore.ModelBases;
+
+namespace ServerCore.Pages.Components
+{
+    public partial class PlayerClassSelectComponent
+    {
+        public int SelectedPlayerClassID { get; set; }
+        public List<PlayerClass> AllPlayerClasses { get; set; }
+        public List<PlayerClass> AvailablePlayerClasses { get; set; }
+
+        [Parameter]
+        public int UserId { get; set; }
+
+        public TeamMembers CurrentTeamMember { get; set; }
+
+        [Parameter]
+        public EventRole CurrentUserEventRole { get; set; }
+
+        [Parameter]
+        public int EventId { get; set; }
+        Event Event { get; set; }
+
+        [Parameter]
+        public bool IsTempClass { get; set; }
+
+        public const int NoClassSetValue = PlayerClassHelper.NoClassSetValue;
+
+        PuzzleServerContext context
+        {
+            get
+            {
+                return Service;
+            }
+        }
+
+        protected override async Task OnParametersSetAsync()
+        {
+            Event = await context.Events.FindAsync(EventId);
+            CurrentTeamMember = await UserEventHelper.GetTeamMemberForPlayer(context, Event, UserId);
+            AllPlayerClasses = await PlayerClassHelper.GetAllPlayerClassesSorted(context, EventId);
+
+            // If a player doesn't have a class in the DB, then set the bound value to the ID for "no class selected"
+            if (CurrentTeamMember.Class == null)
+            {
+                SelectedPlayerClassID = NoClassSetValue;
+            }
+
+            if (IsTempClass && CurrentTeamMember.TemporaryClass != null)
```

Comment:
As good a place as any to say that I don't understand the need for, or meaning of, TemporaryClass at all and do not consider myself to have reviewed its use.

Reply:
That's fair, its use isn't clear without context about the uses for player classes.